core/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -402,12 +368,16 @@ final void innerMoveToSecondPhase() throws Exception {
                 sb.append(result.shardTarget());
             }
 
-            logger.trace("Moving to second phase, based on results from: {} (cluster state version: {})", sb, clusterState.version());
+            logger.trace("Moving to second phase, based on results from: {} (cluster state version: {})", sb, clusterStateVersion);
         }
         moveToSecondPhase();
     }
 
     protected abstract void moveToSecondPhase() throws Exception;
 
     protected abstract String firstPhaseName();
+
+    protected Executor getExecutor() {
+        return threadPool.executor(ThreadPool.Names.SEARCH);
```

Comment:
Maybe pass the `Executor` in to the ctor instead of `ThreadPool`? You could keep this method so you can still override it in tests to throw.